api/src/main/java/org/apache/iceberg/catalog/SupportsNamespaces.java

```diff
@@ -0,0 +1,121 @@
+   * @throws UnsupportedOperationException If create is not a supported operation
+   */
+  default void createNamespace(Namespace namespace) {
+    createNamespace(namespace, ImmutableMap.of());
+  }
+
+  /**
+   * Create a namespace in the catalog.
+   *
+   * @param namespace a multi-part namespace
+   * @param metadata a string Map of properties for the given namespace
+   * @throws AlreadyExistsException If the namespace already exists
+   * @throws UnsupportedOperationException If create is not a supported operation
+   */
+  void createNamespace(Namespace namespace, Map<String, String> metadata);
+
+  /**
+   * List top-level namespaces from the catalog.
+   * <p>
+   * If an object such as a table, view, or function exists, its parent namespaces must also exist
+   * and must be returned by this discovery method. For example, if table a.b.t exists, this method
+   * must return ["a"] in the result array.
+   *
+   * @return an List of namespace {@link Namespace} names
+   */
+  default List<Namespace> listNamespaces() {
+    return listNamespaces(Namespace.empty());
+  }
+
+  /**
+   * List  namespaces from the namespace.
+   * <p>
+   * For example, if table a.b.t exists, use 'SELECT NAMESPACE IN a' this method
+   * must return Namepace.of("a","b") {@link Namespace}.
+   *
+   * @return an List of namespace {@link Namespace} names
+   * @throws NoSuchNamespaceException If the namespace does not exist (optional)
+   */
+  List<Namespace> listNamespaces(Namespace namespace) throws NoSuchNamespaceException;
+
+  /**
+   * Load metadata properties for a namespace.
+   *
+   * @param namespace a Namespace.of(name) {@link Namespace}
+   * @return a string map of properties for the given namespace
+   * @throws NoSuchNamespaceException If the namespace does not exist (optional)
+   */
+  Map<String, String> loadNamespaceMetadata(Namespace namespace) throws NoSuchNamespaceException;
+
+  /**
+   * Drop namespace, while the namespace haven't table or sub namespace will return true.
+   *
+   * @param namespace a Namespace.of(name) {@link Namespace}
+   * @return true while drop success.
+   * @throws NoSuchNamespaceException If the namespace does not exist (optional)
+   */
+  boolean dropNamespace(Namespace namespace) throws NoSuchNamespaceException;
+
+  /**
+   * Apply a set of metadata changes to a namespace in the catalog.
+   *
+   * @param namespace a Namespace.of(name) {@link Namespace}
+   * @param changes a collection of changes to apply to the namespace
+   * @throws NoSuchNamespaceException If the namespace does not exist (optional)
+   * @throws UnsupportedOperationException If namespace properties are not supported
+   */
+  boolean alterNamespace(Namespace namespace, NamespaceChange... changes) throws NoSuchNamespaceException;
+
+  /**
+   * Checks whether the Namespace exists.
+   *
+   * @return true if the Namespace exists, false otherwise
+   */
+  boolean existsNamespace(Namespace namespace);
```

Comment:
This should be named `namespaceExists` to match the table method `tableExists`.

Reply:
I think this should have a default implementation that loads the namespace. If load returns, this should return true. Otherwise, this should return false. Like this:

```java
  default boolean namespaceExists(Namespace namespace) {
    try {
      loadNamespaceMetadata(namespace);
      return true;
    } catch (NoSuchNamespaceException e) {
      return false;
    }
  }
```

Reply:
Thanks, have fixed.